book/src/dev/zebra-dependencies-for-audit.md

```diff
@@ -0,0 +1,150 @@
+| [**zebra-script**](https://github.com/ZcashFoundation/zebra/tree/main/zebra-script) | [1.0.0-rc.0](https://github.com/ZcashFoundation/zebra/tree/v1.0.0-rc.0/zebra-script) |
+| [**zebra-state**](https://github.com/ZcashFoundation/zebra/tree/main/zebra-state) | [1.0.0-rc.0](https://github.com/ZcashFoundation/zebra/tree/v1.0.0-rc.0/zebra-state) |
+| [**zebrad**](https://github.com/ZcashFoundation/zebra/tree/main/zebrad) | [1.0.0-rc.0](https://github.com/ZcashFoundation/zebra/tree/v1.0.0-rc.0/zebrad) |
+| [<i>zebra-utils</i>](https://github.com/ZcashFoundation/zebra/tree/main/zebra-utils) | [1.0.0-rc.0](https://github.com/ZcashFoundation/zebra/tree/v1.0.0-rc.0/zebra-utils) | <i>Only [zebra-checkpoints](https://github.com/ZcashFoundation/zebra/tree/v1.0.0-rc.0/zebra-utils/src/bin/zebra-checkpoints) utility needs to be audited.</i>
+
+## Zcash/ZF dependencies
+
+| Name | Version | Audited | Notes
+|------| --------|-------- | -----
+| [ed25519-zebra](https://github.com/ZcashFoundation/ed25519-zebra) | [3.0.0](https://github.com/ZcashFoundation/ed25519-zebra/releases/tag/3.0.0) | |
+| [equihash](https://github.com/zcash/librustzcash) | [0.2.0](https://github.com/zcash/librustzcash/releases/tag/0.2.0) | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) |
+| [halo2_proofs](https://github.com/zcash/halo2) | [0.2.0](https://github.com/zcash/halo2/tree/halo2_proofs-0.2.0) | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) [mary](https://z.cash/halo2-audit/) |
+| [incrementalmerkletree](https://github.com/zcash/incrementalmerkletree) | [0.3.0](https://github.com/zcash/incrementalmerkletree/releases/tag/v0.3.0) | |  
+| [zcash_encoding](https://github.com/zcash/librustzcash) | [0.1.0](https://github.com/zcash/librustzcash/releases/tag/0.1.0) | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) | 
+| [zcash_history](https://github.com/zcash/librustzcash) | 0.3.0 | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) |
+| [zcash_note_encryption](https://github.com/zcash/librustzcash) | [0.1.0](https://github.com/zcash/librustzcash/releases/tag/0.1.0) | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) |
+| [zcash_primitives](https://github.com/zcash/librustzcash) | 0.7.0 | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) |
+| [redjubjub](https://github.com/ZcashFoundation/redjubjub) | [0.5.0](https://github.com/ZcashFoundation/redjubjub/releases/tag/0.5.0) | [jp](https://github.com/ZcashFoundation/redjubjub/raw/main/zcash-frost-audit-report-20210323.pdf) |
+| [orchard](https://github.com/zcash/orchard) | [0.2.0](https://github.com/zcash/orchard/releases/tag/0.2.0) | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) |
+| [zcash_proofs](https://github.com/zcash/librustzcash.git?rev=ca84123038b64f0f4aa5615f7cf224fbf7ece766#ca841230) | [0.7.1](https://github.com/zcash/librustzcash.git?rev=ca84123038b64f0f4aa5615f7cf224fbf7ece766#ca841230/releases/tag/v0.7.1) | [qedit](https://hackmd.io/@qedit/zcash-nu5-audit) |
```

Comment:
We might want to audit @upbqdn's latest changes to the `zcash_proofs` downloader, since `zcashd` doesn't use them.

@upbqdn can you provide a git commit link and directory/files we need to audit?

Reply:
It looks like this will be in the upcoming `zcash_proofs` release.

@mpguerra do we want to upgrade our ECC dependencies before we tag the next release candidate?
That would get rid of duplicate ECC dependencies (#3831), and make it easier to specify dependency versions.

(Only the `zcash_proofs` Zcash parameter download code might be in scope.)

Reply:
yes, I'm not in a rush to tag the next release candidate so we can wait for the ECC dependency upgrade to happen.

Reply:
> @upbqdn can you provide a git commit link and directory/files we need to audit?

The PR consisted of three commits. The last one was https://github.com/zcash/librustzcash/commit/75b403686261a055b3061e7d168b3a4c86378ed4. The `librustzcash` repo doesn't squash commits in PRs.

The files are https://github.com/zcash/librustzcash/blob/75b403686261a055b3061e7d168b3a4c86378ed4/zcash_proofs/src/downloadreader.rs, and https://github.com/zcash/librustzcash/tree/75b403686261a055b3061e7d168b3a4c86378ed4/zcash_proofs/src/lib.rs.
I pointed both files at the last commits from the PR that contains the fix.

Reply:
Should I submit a PR that edits the tables directly instead of the comment above?

Reply:
> Should I submit a PR that edits the tables directly instead of the comment above? I'm not sure what would be the best way to embed the information from the comment into the table, though.

I made an attempt to add this to the table in https://github.com/ZcashFoundation/zebra/pull/5471, let me know or submit a PR that edits the tables directly if that doesn't work.

Reply:
> Should I submit a PR that edits the tables directly instead of the comment above? I'm not sure what would be the best way to embed the information from the comment into the table, though.

I made an attempt to add this to the table in https://github.com/ZcashFoundation/zebra/pull/5471, let me know or submit a PR that edits the tables directly if that doesn't work.